### ISSUE TITLE
Feature request: search local config file recursively in parent directories

### ISSUE THREAD

**Issue by Niluge-KiWi** [svent/sift#3]
It would be great to be able to set a sift config file for a whole subtree, and use sift only in a subdirectory of it:

``` shell
$ cat ~/tree/.sift.conf
{"ShowLineNumbers": true}
$ cd ~/tree/subdir
$ sift test
file:1: test
```

Currently sift only searches the local config file in the current directory, which forces us to either put config in the global config file (thus applying it too broadly), or duplicating it in each subdir where we want the special config (not maintainable).

**Comment by GeertJohan** [comment]
:+1:
This would be great to have sift configs defined per project. I would like to suggest to name this file `.siftconfig` instead of `.sift.conf`, its more in line with the comon file names these days (.gitignore, etc)

**Comment by svent** [comment]
Thanks for the suggestion - I already thought about this and it really seems that this is a behavior people would expect and that is definitely useful.
I will implement this in the next version.

Renaming the config file would break existing installations, so I will not change it for now.

**Comment by svent** [comment]
I just implemented this and released version 0.3.3.
